tracer/src/Datadog.Trace.Manual/FeatureFlags/FeatureFlagsSdk.cs

```diff
@@ -0,0 +1,48 @@
+// <copyright file="FeatureFlagsSdk.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using Datadog.Trace.SourceGenerators;
+
+namespace Datadog.Trace.FeatureFlags;
+
+/// <summary>
+/// Functions to retrieve FeatureFlags from server
+/// </summary>
+[EditorBrowsable(EditorBrowsableState.Never)]
+[Browsable(false)]
+public static class FeatureFlagsSdk
+{
+    /// <summary> Gets a value indicating wether FeatureFlags framework is available or not </summary>
+    /// <returns> True if FeatureFlagsSDK is instrumented </returns>
+    [Instrumented]
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static bool IsAvailable() => false;
+
+    /// <summary> Installs an event handler to be fired when a new config has been received </summary>
+    /// <param name="onNewConfig"> Action to be called when the event is fired </param>
+    [Instrumented]
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static void RegisterOnNewConfigEventHandler(Action onNewConfig)
+    {
+    }
+
+    /// <summary>
+    /// Sets the details of a successful login on the local root span
+    /// </summary>
+    /// <returns> Returns the evaluation result </returns>
+    /// <param name="key">The feature flag key to evaluate</param>
+    /// <param name="targetType">The desired result type</param>
+    /// <param name="defaultValue">The default value</param>
+    /// <param name="context">The evaluation context</param>
+    [Instrumented]
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static IEvaluation? Evaluate(string key, ValueType targetType, object? defaultValue, IEvaluationContext? context)
```

Comment:
I wonder about this `IEvaluationContext` 🤔 The _trouble_ with us accepting an `IEvaluationContext` object here, is that the user can implement their own `IEvaluationContext` and pass it in, which in the past has been problematic 🤔 I would suggest two things:

- Force passing in a concrete `EvaluationContext` object
- Don't instrument that method, instrument a method it _delegates_ too, which only uses shared BCL types. It would give us more future proofing (e.g. we could change the public API without breaking ourselves) and avoid a layer of duck typing 🤔 


e.g.


```suggestion
    // NOT instrumented
    public static IEvaluation? Evaluate(string key, ValueType targetType, object? defaultValue, EvaluationContext? context)
        => Evaluate(key, targetType, defaultValue, context?.TargetingKey, context?.Attributes);

    [Instrumented]
    [MethodImpl(MethodImplOptions.NoInlining)]
    public static IEvaluation? Evaluate(string key, ValueType targetType, object? defaultValue, string? targetingKey, IDictionary<string, object?>? attributes)
```

Reply:
Noted (and TY for the change)